### ISSUE TITLE
http.Client round-robin across persistent connections

### ISSUE THREAD

**Issue by CAFxX** [golang/go#34511]
When connecting directly to a service spread across multiple hosts with DNS "load-balancing" it would be ideal if `http.Client` could round-robin requests over persistent connections to the multiple hosts listed in the A records.

Right now, if you want to ensure that load is balanced the only practical option is to disable HTTP keep alives.

One way this could be done is to have idle connection pools for each different A record, and do a DNS resolution for every request to decide which pool to use for that request.

This roundrobin mode should likely be opt-in.

**Comment by agnivade** [comment]
@bradfitz @fraenkel

**Comment by bradfitz** [comment]
Agreed. I've been thinking about things like this as part of #23707 and making the connection pool interface more explicit/configurable. Currently it's a pretty hidden detail.

**Comment by frittentheke** [comment]
@bradfitz could you kindly give an update if connection pooling and doing round-robin over multiple connections (to multiple IPs even) is something that can be done?

**Comment by frittentheke** [comment]
@bradfitz could you kindly give an update if you are working on load-balancing / round-robing requests at all? https://github.com/golang/go/issues/23707 seems dead.

@CAFxX I know you did implement https://github.com/CAFxX/balancer, but do you have any plans to get (something like) this into upstream net/http then? Would be great to have.
